python/pyiceberg/expressions/base.py

```diff
@@ -867,3 +875,196 @@ def visit_unbound_predicate(self, predicate) -> BooleanExpression:
 
     def visit_bound_predicate(self, predicate) -> BooleanExpression:
         return predicate
+
+
+ROWS_MIGHT_MATCH = True
+ROWS_CANNOT_MATCH = False
+IN_PREDICATE_LIMIT = 200
+
+
+def _from_byte_buffer(field_type: IcebergType, val: bytes):
+    if not isinstance(field_type, PrimitiveType):
+        raise ValueError(f"Expected a PrimitiveType, got: {type(field_type)}")
+    return from_bytes(field_type, val)
+
+
+class ManifestEvaluator(BoundBooleanExpressionVisitor[bool]):
+    partition_fields: list[PartitionFieldSummary]
+    partition_filter: BooleanExpression
+
+    def __init__(self, schema: Schema, partition_filter: UnboundPredicate, case_sensitive: bool = True):
```

Comment:
It is unlikely that this is going to be a `Schema` because it will be the partition type produced by #5929.

I think what we want instead is to pass the Schema and the PartitionSpec to produce the type that the expression should be bound to.

Confusing a data filter (bound to Schema) with a partition filter (bound to partition type) is something that we want to avoid and is why there are static factory methods in the Java code. This is okay, but we will need to be careful here.

Reply:
Yes, this was a temp solution to test the binding 👍🏻 Will be replaced once #5929 has been merged